02_configure_host.sh

```diff
@@ -312,6 +312,15 @@ mkdir -p "${M3PATH}"
 detect_mismatch "${BMO_LOCAL_IMAGE:-}" "${BMOPATH}"
 clone_repo "${BMOREPO}" "${BMOBRANCH}" "${BMOPATH}" "${BMOCOMMIT}"
 
+# We are fading away from deploy.sh, in favor of deploy-cli. In the meantime,
+# as deploy-cli is not available in all BMO branches, and as deploy.sh assumes
+# to be in ${BMOPATH}/tools, we compile deploy-cli and put it in deploy.sh place,
+# if it's available, and use deploy.sh otherwise.
+
+pushd "${BMOPATH}"
+make deploy-cli && chmod +x tools/bin/deploy-cli && rm tools/deploy.sh && cp tools/bin/deploy-cli tools/deploy.sh
```

Comment:
What happens when we run this in BMO 0.8 or older? It will just exit the script with failure?

Reply:
There is a lesson to be learned here, just don't know what lesson that would be. This definitely should exit, as there is "set -e", make is returning 2 for failure, and && does not trigger pipefail shenanigans.

Anyways, let's write it properly, ie.

```suggestion
pushd "${BMOPATH}"
if make deploy-cli; then 
    chmod +x tools/bin/deploy-cli
    rm tools/deploy.sh
    cp tools/bin/deploy-cli tools/deploy.sh
fi
popd
```